File: project/.sdk/tm/lua/utility/done.lua
-- ProjectName SDK utility: done

local function done_util(ctx)
  if ctx.ctrl.explain ~= nil then
    ctx.ctrl.explain = ctx.utility.clean(ctx, ctx.ctrl.explain)
    local explain_result = ctx.ctrl.explain["result"]
    if type(explain_result) == "table" then
      explain_result["err"] = nil
    end
  end

  if ctx.result ~= nil and ctx.result.ok then
    return ctx.result.resdata, nil
  end

  return ctx.utility.make_error(ctx, nil)
end

return done_util

File: project/.sdk/tm/py/utility/done.py
# ProjectName SDK utility: done


def done_util(ctx):
    if ctx.ctrl.explain is not None:
        ctx.ctrl.explain = ctx.utility.clean(ctx, ctx.ctrl.explain)
        explain_result = ctx.ctrl.explain.get("result") if isinstance(ctx.ctrl.explain, dict) else None
        if isinstance(explain_result, dict):
            explain_result.pop("err", None)

    if ctx.result is not None and ctx.result.ok:
        return ctx.result.resdata, None

    return ctx.utility.make_error(ctx, None)
